### PR TITLE
feat: UI Test for edit period duration

### PR DESCRIPTION
**Pull request by alainajaffer**
## Description

UI test to edit the product period duration in Cypress test

## Testing instructions

Run myFares.cy tests all should pass

### CODE REVIEW

File: repos/fdbt-site/cypress_tests/cypress/support/steps.ts [hunk 1]
```diff
@@ -396,7 +396,21 @@ export const editProductExpiry = () => {
...
     randomlyChooseProductPeriodValidity();
     continueButtonClick();
-    clickElementByText('Back');    
+    clickElementByText('Back');
+};
+
+export const editProductDuration = () => {
+    clickElementById('period-duration-link');
+    const productDuration = getRandomNumber(1, 10);
+    clickElementById('edit-period-duration-period-duration-quantity').clear().type(productDuration.toString());
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/cypress_tests/cypress/support/steps.ts:405]: can we have the name as edit-period-duration-quantity ?

File: repos/fdbt-site/cypress_tests/cypress/support/steps.ts [hunk 2]
```diff
@@ -396,7 +396,21 @@ export const editProductExpiry = () => {
...
+    selectRandomOptionFromDropDown('edit-period-duration-period-duration-unit');
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/cypress_tests/cypress/support/steps.ts:406]: period-duration-period-duration-unit ... 
period-duration is repeating

File: repos/fdbt-site/src/pages/api/editPeriodDuration.ts [hunk 3]
```diff
@@ -37,8 +37,8 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
         if (!ticket || !matchingJsonMetaData) {
             throw new Error('Ticket information cannot be undefined');
         }
-        const productDurationValue = req.body.multipleProductDurationInput;
-        const productDurationUnit = req.body.multipleProductDurationUnitsInput;
+        const productDurationValue = req.body.editPeriodDurationInput;
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/editPeriodDuration.ts:40]: shouldn't we use periodDurationInput ?

File: repos/fdbt-site/src/pages/editPeriodDuration.tsx [hunk 4]
```diff
@@ -45,11 +45,11 @@ const EditPeriodDuration = ({
                     <ExpirySelector
                         defaultDuration={productDurationValue || undefined}
                         defaultUnit={(productDurationUnit as ExpiryUnit) || undefined}
-                        quantityName={'multipleProductDurationInput'}
-                        quantityId={'product-details-period-duration-quantity'}
-                        hintId="product-period-duration-hint"
-                        unitName={'multipleProductDurationUnitsInput'}
-                        unitId={`product-details-period-duration-unit`}
+                        quantityName={'editPeriodDurationInput'}
+                        quantityId={'edit-period-duration-period-duration-quantity'}
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/editPeriodDuration.tsx:49]: change the name please to remove doubling period-duration

File: repos/fdbt-site/src/pages/editPeriodDuration.tsx [hunk 5]
```diff
@@ -45,11 +45,11 @@ const EditPeriodDuration = ({
...
+                        hintId="edit-period-duration-period-duration-hint"
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/editPeriodDuration.tsx:50]: period-duration is doubling

File: repos/fdbt-site/src/pages/editPeriodDuration.tsx [hunk 6]
```diff
@@ -45,11 +45,11 @@ const EditPeriodDuration = ({
...
+                        unitName={'editPeriodDurationUnitsInput'}
+                        unitId={'edit-period-duration-period-duration-unit'}
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/editPeriodDuration.tsx:52]: period-duration-

File: repos/fdbt-site/tests/pages/__snapshots__/editPeriodDuration.test.tsx.snap [hunk 7]
```diff
@@ -41,12 +41,12 @@ exports[`pages editPeriodDuration should render editPeriodDuration page correctl
         defaultUnit="week"
         errors={Array []}
         hideFormGroupError={true}
-        hintId="product-period-duration-hint"
-        quantityId="product-details-period-duration-quantity"
-        quantityName="multipleProductDurationInput"
+        hintId="edit-period-duration-period-duration-hint"
+        quantityId="edit-period-duration-period-duration-quantity"
+        quantityName="editPeriodDurationInput"
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/tests/pages/__snapshots__/editPeriodDuration.test.tsx.snap:46]: i think we can remove the edit keyword from here

File: repos/fdbt-site/tests/pages/api/editPeriodDuration.test.ts [hunk 8]
```diff
@@ -18,7 +18,7 @@ describe('editPeriodDuration tests', () => {
     it('should update a products period duration and redirect back to products/productDetails', async () => {
         const { req, res } = getMockRequestAndResponse({
             cookieValues: {},
-            body: { multipleProductDurationInput: '6', multipleProductDurationUnitsInput: 'week' },
+            body: { editPeriodDurationInput: '6', editPeriodDurationUnitsInput: 'week' },
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/tests/pages/api/editPeriodDuration.test.ts:21]: remove the edit keyword

File: repos/fdbt-site/tests/pages/__snapshots__/editPeriodDuration.test.tsx.snap [hunk 9]
```diff
@@ -41,12 +41,12 @@ exports[`pages editPeriodDuration should render editPeriodDuration page correctl
...
         school={false}
-        unitId="product-details-period-duration-unit"
-        unitName="multipleProductDurationUnitsInput"
+        unitId="edit-period-duration-period-duration-unit"
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/tests/pages/__snapshots__/editPeriodDuration.test.tsx.snap:48]: remove the doubling of period-duration

File: repos/fdbt-site/cypress_tests/cypress/support/helpers.ts [hunk 10]
```diff
@@ -592,43 +595,51 @@ export const addOtherProductsIfNotPresent = (): void => {
     clickElementByText('Other products');
 
     let numberOfPeriodProducts = 0;
+    cy.wrap(numberOfPeriodProducts).as('numberOfPeriodProducts');
     let numberOfFlatFareProducts = 0;
+    cy.wrap(numberOfFlatFareProducts).as('numberOfFlatFareProducts');
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/cypress_tests/cypress/support/helpers.ts:600]: Will these not be stored as zero?

Comment (RobCattonKPMG) [reply repos/fdbt-site/cypress_tests/cypress/support/helpers.ts:600]: Ah - we store them just incase they're never incremented.

**Review by RobCattonKPMG — APPROVED** [review]

**Pull request merged**